test/php/Nominatim/ParameterParserTest.php

```diff
@@ -0,0 +1,214 @@
+                                        'int3' => '-1',
+                                        'int4' => '',
+                                        'int5' => 0
+                                       ]);
+
+        $this->assertEquals(false, $oParams->getInt('non-exists'));
+        $this->assertEquals(999, $oParams->getInt('non-exists', 999));
+        $this->assertEquals(5, $oParams->getInt('int1'));
+
+        try {
+            $this->assertEquals(false, $oParams->getInt('int2'));
+        } catch (Exception $e) {
+            $this->assertEquals($e->getMessage(), "Integer number expected for parameter 'int2'");
+        }
+        $this->assertEquals(-1, $oParams->getInt('int3'));
+        $this->assertEquals(false, $oParams->getInt('int4'));
+        $this->assertEquals(false, $oParams->getInt('int5')); // FIXME: should be 0 instead?
+    }
+
+
+    public function testGetFloat()
+    {
+        $oParams = new ParameterParser([
+                                        'float1' => '1.0',
+                                        'float2' => '-5',
+                                        'float3' => '-55.',
+                                        'float4' => 'a',
+                                        'float5' => '',
+                                        'float6' => 0
+                                       ]);
+
+        $this->assertEquals(false, $oParams->getFloat('non-exists'));
+        $this->assertEquals(999, $oParams->getFloat('non-exists', 999));
+        $this->assertEquals(1, $oParams->getFloat('float1'));
+        $this->assertEquals(-5, $oParams->getFloat('float2'));
+
+        try {
+            $this->assertEquals(false, $oParams->getFloat('float3'));
+        } catch (Exception $e) {
+            $this->assertEquals($e->getMessage(), "Floating-point number expected for parameter 'float3'");
+        }
+
+        try {
+            $this->assertEquals(false, $oParams->getFloat('float4'));
+        } catch (Exception $e) {
+            $this->assertEquals($e->getMessage(), "Floating-point number expected for parameter 'float4'");
+        }
+        $this->assertEquals(false, $oParams->getFloat('float5'));
+        $this->assertEquals(false, $oParams->getFloat('float6')); // FIXME: should be 0 instead?
+    }
+
+
+
+    public function testGetString()
+    {
+        $oParams = new ParameterParser([
+                                        'str1' => 'abc',
+                                        'str2' => '',
+                                        'str3' => '0'
+                                       ]);
+
+        $this->assertEquals(false, $oParams->getString('non-exists'));
+        $this->assertEquals('default', $oParams->getString('non-exists', 'default'));
+        $this->assertEquals('abc', $oParams->getString('str1'));
+        $this->assertEquals(false, $oParams->getStringList('str2'));
+        $this->assertEquals(false, $oParams->getStringList('str3')); // FIXME: should be 0 instead?
+    }
+
+
+
```

Comment:
Is phpcs happy with so many empty lines?

Reply:
We disabled the `Squiz.WhiteSpace.FunctionSpacing.After` rule because it complained about having 1 empty line after function closing block (afai remember). Looks like enabling it requires only two fixes. I'll address that in a separate PR.